src/core/document.js

```diff
@@ -935,6 +938,65 @@ class PDFDocument {
       ? this.catalog.cleanup(manuallyTriggered)
       : clearPrimitiveCaches();
   }
+
+  _collectFieldObjects(name, fieldRef, promises) {
+    const field = this.xref.fetchIfRef(fieldRef);
+    if (field.has("T")) {
+      const partName = stringToPDFString(field.get("T"));
+      if (name === "") {
+        name = partName;
+      } else {
+        name = `${name}.${partName}`;
+      }
+    }
+
+    if (!(name in promises)) {
+      promises[name] = [];
+    }
+    promises[name].push(
+      AnnotationFactory.create(
+        this.xref,
+        fieldRef,
+        this.pdfManager,
+        this._localIdFactory
+      )
+        .then(annotation => annotation.getFieldObject())
+        .catch(function (reason) {
+          warn(`_collectFieldObjects: "${reason}".`);
+          return null;
+        })
+    );
+
+    if (field.has("Kids")) {
+      const kids = field.get("Kids");
+      for (const kid of kids) {
+        this._collectFieldObjects(name, kid, promises);
+      }
+    }
+  }
+
+  async getFieldObjects() {
```

Comment:
Any reason to not simply make this a getter instead?
I.e. `async get fieldObjects() {` instead, since then you could also `shadow` the return values similar to most existing methods in `PDFDocument`.

Reply:
I did that but `async get foo` is not allowed and then I thought that it was useless to keep in memory the `fieldObjects` because we'll use it only one time.